src/Traits/CopiesToBuildDirectory.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace Native\Electron\Traits;
+
+use RecursiveCallbackFilterIterator;
+use RecursiveDirectoryIterator;
+use RecursiveIteratorIterator;
+
+use function Laravel\Prompts\intro;
+use function Laravel\Prompts\note;
+
+trait CopiesToBuildDirectory
+{
+    abstract protected function buildPath(): string;
+
+    protected function copyToBuildDirectory()
+    {
+        intro('Copying App to build directory...');
+
+        $sourcePath = base_path();
+        $buildPath = $this->buildPath();
+
+        $this->removeDirectory($buildPath);
+        mkdir($buildPath, 0755, true);
+
+        // A filtered iterator that will exclude files matching our skip patterns
+        $directory = new RecursiveDirectoryIterator($sourcePath, RecursiveDirectoryIterator::SKIP_DOTS | RecursiveDirectoryIterator::FOLLOW_SYMLINKS);
+
+        $filter = new RecursiveCallbackFilterIterator($directory, function ($current) {
+            $relativePath = substr($current->getPathname(), strlen(base_path()) + 1);
+            $patterns = config('nativephp.cleanup_exclude_files') + static::CLEANUP_PATTERNS;
+
+            // Check each skip pattern against the current file/directory
+            foreach ($patterns as $pattern) {
+                // fnmatch supports glob patterns like "*.txt" or "cache/*"
+                if (fnmatch($pattern, $relativePath)) {
+                    return false;
+                }
+            }
+
+            return true;
+        });
+
+        // Now we walk all directories & files and copy them over accordingly
+        $iterator = new RecursiveIteratorIterator($filter, RecursiveIteratorIterator::SELF_FIRST);
+
+        foreach ($iterator as $item) {
+            $target = $buildPath.DIRECTORY_SEPARATOR.substr($item->getPathname(), strlen($sourcePath) + 1);
+
+            if ($item->isDir()) {
+                if (! is_dir($target)) {
+                    mkdir($target, 0755, true);
+                }
+
+                continue;
+            }
+
+            copy($item->getPathname(), $target);
+        }
+
+        $this->keepRequiredDirectories();
+
+        note('App copied');
+    }
+
+    private function removeDirectory($path)
+    {
+        if (! file_exists($path)) {
+            return;
+        }
+
+        if (is_file($path) || is_link($path)) {
+            return unlink($path);
+        }
+
+        $dir = dir($path);
+        while (false !== $entry = $dir->read()) {
+            if ($entry == '.' || $entry == '..') {
+                continue;
+            }
+
+            $this->removeDirectory($path.DIRECTORY_SEPARATOR.$entry);
+        }
+
+        // Clean up
+        $dir->close();
+        rmdir($path);
+    }
+
+    private function keepRequiredDirectories()
```

Comment:
I'm not 100% sure why this is needed... the `storage` directory that the application looks for should be the one in `AppData`... would be good to be clear on _why_ this was / still is necessary

Reply:
Good point. I ported this part over from the current implementation. Will check if this is still needed

Reply:
It seems this is still required. When starting the app for the first time, these folders will not be created unless a file is in there

Reply:
Ok cools. Can you add a comment in the code here explaining that?

Reply:
Copied the original comment here: https://github.com/NativePHP/electron/blob/a2633ac4dae53288524d1faea84e199aedb4fff6/src/Traits/CopiesToBuildDirectory.php#L71